operators/pkg/controller/kibana/config/reconciler.go

```diff
@@ -0,0 +1,68 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package config
+
+import (
+	"reflect"
+
+	"github.com/elastic/cloud-on-k8s/operators/pkg/apis/kibana/v1alpha1"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/common/reconciler"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/common/settings"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/kibana/label"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/utils/k8s"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+// ReconcileConfigSecret reconciles the expected Kibana config secret for the given Kibana resource.
+// This managed secret is mounted into each pod of the Kibana deployment.
+func ReconcileConfigSecret(client k8s.Client, kb v1alpha1.Kibana, kbSettings *settings.CanonicalConfig) error {
+	settingsYamlBytes, err := kbSettings.Render()
+	if err != nil {
+		return err
+	}
+	expected := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: kb.Namespace,
+			Name:      SecretName(kb),
+			Labels: map[string]string{
+				label.KibanaNameLabelName: kb.Name,
+			},
+		},
+		Data: map[string][]byte{
+			settingsFilename: settingsYamlBytes,
+		},
+	}
+	reconciled := corev1.Secret{}
+	if err := reconciler.ReconcileResource(reconciler.Params{
+		Client:     client,
+		Scheme:     scheme.Scheme,
+		Owner:      &kb,
+		Expected:   &expected,
+		Reconciled: &reconciled,
+		NeedsUpdate: func() bool {
+			return !reflect.DeepEqual(reconciled.Data, expected.Data)
+		},
+		UpdateReconciled: func() {
+			reconciled.Data = expected.Data
+		},
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+// getConfigSecret returns the Kibana config secret for the given Kibana resource.
+func getConfigSecret(client k8s.Client, kb v1alpha1.Kibana) (*corev1.Secret, error) {
```

Comment:
Naming: should this be called auth secret? Config secret implies for me the secret through which we mount the config files. But this seems to contain the Es auth info?

Reply:
Absolutely. I'll rename it.